### PR TITLE
esp32&esp32c3/wifi: Support specific channel and bssid scan

### PR DESCRIPTION
**Pull request by cwespressif**
## Summary

Support specific channel and bssid scan

## Testing

- Tested with ESP32-Devkitc or ESP32-C3-Devkitc

`./tools/configure esp32c3-devkit:wapi` or `./tools/configure esp32c3-devkit:sta_softap`
`./tools/configure esp32-devkitc:wapi` or `./tools/configure esp32-devkitc:sta_softap`

### CODE REVIEW

File: arch/xtensa/src/esp32/esp32_wifi_utils.c [hunk 1]
```diff
@@ -143,6 +153,36 @@ int esp_wifi_start_scan(struct iwreq *iwr)
...
+              g_channel_num = req->num_channels;
+            }
+          else
+            {
+              DEBUGASSERT(req->num_channels <= CHANNEL_MAX_NUM);
+              for (i = 0; i < req->num_channels; i++)
+                {
+                  g_channel_num = req->num_channels;
+                  g_channel_list[i] = req->channel_list[i].m;
+                }
```
Comment (Ouss4) [inline arch/xtensa/src/esp32/esp32_wifi_utils.c:174]:
```suggestion
              DEBUGASSERT(req->num_channels <= CHANNEL_MAX_NUM);
              
              g_channel_num = req->num_channels;
              for (i = 0; i < req->num_channels; i++)
                {
                  g_channel_list[i] = req->channel_list[i].m;
                }
```

File: arch/risc-v/src/esp32c3/esp32c3_wifi_utils.c [hunk 2]
```diff
@@ -143,6 +154,36 @@ int esp_wifi_start_scan(struct iwreq *iwr)
...
+              g_channel_num = req->num_channels;
+            }
+          else
+            {
+              DEBUGASSERT(req->num_channels <= CHANNEL_MAX_NUM);
+              for (i = 0; i < req->num_channels; i++)
+                {
+                  g_channel_num = req->num_channels;
+                  g_channel_list[i] = req->channel_list[i].m;
+                }
```
Comment (Ouss4) [inline arch/risc-v/src/esp32c3/esp32c3_wifi_utils.c:175]:
```suggestion
              DEBUGASSERT(req->num_channels <= CHANNEL_MAX_NUM);
              
              g_channel_num = req->num_channels;
              for (i = 0; i < req->num_channels; i++)
                {
                  g_channel_list[i] = req->channel_list[i].m;
                }
```

File: arch/xtensa/src/esp32/esp32_wifi_utils.c (hunk 1 above)
Comment (cwespressif) [reply arch/xtensa/src/esp32/esp32_wifi_utils.c]: Done

File: arch/risc-v/src/esp32c3/esp32c3_wifi_utils.c (hunk 2 above)
Comment (cwespressif) [reply arch/risc-v/src/esp32c3/esp32c3_wifi_utils.c]: Done

**Review by Ouss4 — APPROVED** [review]

**Pull request merged**